package.json

```diff
@@ -43,9 +43,9 @@
     "test": "jest",
     "test:watch": "npm test -- --watch",
     "test:cov": "npm test -- --coverage",
-    "build": "rollup -c",
-    "prepare": "npm run clean && npm run check-types && npm run format:check && npm run lint && npm test",
-    "check-types": "tsc",
```

Comment:
Can we keep this with `--noEmit` on it so that we still have a type check as a part of the CI process? I know this happens by proxy of the test suite, but an earlier, more-focused test will make the problems easier to see.

Reply:
`"noEmit": true` is in `tsconfig.json` which is the same thing. Do you want me to add back in the `check-types` command, even though it will basically run `tsc` twice the same way?

Reply:
I would do this:

- Set `declaration: false` in the tsconfig.
- Make `build-types` run `tsc -d`
- Make `check-types` run `tsc --noEmit` to be explicit about it.
- Run `check-types` in the `prepare` script.

Another option: Are we using typescript-eslint? That does a type check as well, so that would be sufficient for checking them.

Reply:
I think I found a solution similar to this, although the declaration: false is unneeded if we do noEmit. The `--emitDeclarationOnly` flag does what we need. take a look when you're free and see if it fits the bill.

I also added `types/` to the `clean` command, as we will definitely want that